policies/api/http/endpoint.go

```diff
@@ -415,3 +415,31 @@ func listDatasetEndpoint(svc policies.Service) endpoint.Endpoint {
 		return res, nil
 	}
 }
+
+func duplicatePolicyEndpoint(svc policies.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(viewResourceReq)
```

Comment:
This should optionally take the new policy name, which if it's passed should be used instead of generating a new name based on the old one